### PR TITLE
DAOS-13672 control: Bump system_ram_reserved to reduce OOM occurrences

### PR DESCRIPTION
**Pull request by tanabarr**
Attempt to reduce the chance of OOM killer terminating an engine
process when maximum pool space is allocated by slightly increasing
the system_ram_reserved default value from 6->16gib.

Some test yaml system_ram_reserved values have been reduced to 6 to
prevent the increase in the default from causing an available memory
check failure on engine start-up in memory constrained (VM)
environments. Increasing the default value should also resolve
DAOS-13918 by providing a larger memory buffer to reduce the chance
of intermittent failures related to this check.

Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by NiuYawei — DISMISSED** [review]

**Review by daltonbohning — DISMISSED** [review]

**Review by NiuYawei — DISMISSED** [review]

**Comment by tanabarr** [comment]
merged DAOS-control: Bump system_ram_reserved to reduce OOM occurrences #12430 into this PR

**Review by knard-intel — DISMISSED** [review]
LGTM with just one typo detail.

**Review by kjacque — DISMISSED** [review]

**Review by NiuYawei — APPROVED** [review]

**Review by kjacque — APPROVED** [review]

**Review by kjacque — APPROVED** [review]

**Review by NiuYawei — APPROVED** [review]

**Review by knard-intel — DISMISSED** [review]

**Comment by tanabarr** [comment]
PR-12430 branch build 16:

2 * PoolCreateTests - DAOS-13967,DAOS-13916
DmgSystemReformatTest - DAOS-13613
DaosServerConfigTest - DAOS-13918
TelemetryPoolSpaceMetrics - DAOS-14016
DmgStorageQuery - DAOS-13687

All failures unrelated to PR and apart from DAOS-14016 all listed in https://daosio.atlassian.net/wiki/spaces/DC/pages/11387633710/Master+Metadata+on+SSD+Daily+Test+Results .

File: src/tests/ftest/server/daos_server_config.yaml
Comment (mchaarawi) [inline src/tests/ftest/server/daos_server_config.yaml:9]: can you please explain why there are instances where in the yaml file you set this to 6 and in others 16?

**Comment by phender** [comment]
There is no test result for the `src/tests/ftest/security/cont_update_acl.yaml` change in build https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12430/16, but the test was run and passed in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12430/14/artifact/Functional%20on%20EL%208/security/cont_update_acl.py with the added `system_ram_reserved` value confirmed to be in use.  Ideally all tests should be run in the last commit, but if testing is broken out across multiple builds it should be noted for the gatekeepers.

**Comment by phender** [comment]
@tanabarr the `1-./telemetry/pool_space_metrics.py:TelemetryPoolSpaceMetrics.test_telemetry_pool_space_metrics` test time out issue (https://daosio.atlassian.net/browse/DAOS-14016) has been seen twice in this PR and has not been seen in daily md-on-ssd-testing builds (https://daosio.atlassian.net/wiki/spaces/DC/pages/11387633710/Master+Metadata+on+SSD+Daily+Test+Results).

This would seem to indicate that some change in this PR causes this test to take longer to execute.  I think the correct course of action would be to have this PR increase the timeout in https://github.com/daos-stack/daos/blob/master/src/tests/ftest/telemetry/telemetry_pool_metrics.yaml#L4.  I suspect 60 seconds should suffice.

File: src/tests/ftest/server/daos_server_config.yaml
Comment (tanabarr) [reply src/tests/ftest/server/daos_server_config.yaml:9]: The instances where the yaml file value have been reduced to 6 were as a result of the bump in the default value causing a memory check failure. This is likely to be in situations where the environment in which the CI test-in-question is memory constrained and so reserving 8gib rather than 6gib doesn't allow a minimum (4gib) RAM-disk to be allocated to the engine after taking into account other memory reservations in the calculation.

Some other tests in CI that are intermittently experiencing memory check failures (e.g. https://daosio.atlassian.net/browse/DAOS-13918) due to RAM usage spikes have been "fixed" in this PR by increasing the `system_ram_reserved` for hardware based tests to 16gib to give a larger buffer which should reduce if not remove the chance of intermittent failures related to this check.

Comment (tanabarr) [reply src/tests/ftest/server/daos_server_config.yaml:9]: description updated

**Comment by tanabarr** [comment]
> @tanabarr the `1-./telemetry/pool_space_metrics.py:TelemetryPoolSpaceMetrics.test_telemetry_pool_space_metrics` test time out issue (https://daosio.atlassian.net/browse/DAOS-14016) has been seen twice in this PR and has not been seen in daily md-on-ssd-testing builds (https://daosio.atlassian.net/wiki/spaces/DC/pages/11387633710/Master+Metadata+on+SSD+Daily+Test+Results).
> 
> This would seem to indicate that some change in this PR causes this test to take longer to execute. I think the correct course of action would be to have this PR increase the timeout in https://github.com/daos-stack/daos/blob/master/src/tests/ftest/telemetry/telemetry_pool_metrics.yaml#L4. I suspect 60 seconds should suffice.

I'm not clear on which part of the PR would increase telemetry test duration but I will push the timeout change as suggested.

**Review by kjacque — DISMISSED** [review]
Fine with me.

**Comment by tanabarr** [comment]
> @tanabarr the `1-./telemetry/pool_space_metrics.py:TelemetryPoolSpaceMetrics.test_telemetry_pool_space_metrics` test time out issue (https://daosio.atlassian.net/browse/DAOS-14016) has been seen twice in this PR and has not been seen in daily md-on-ssd-testing builds (https://daosio.atlassian.net/wiki/spaces/DC/pages/11387633710/Master+Metadata+on+SSD+Daily+Test+Results).
> 
> This would seem to indicate that some change in this PR causes this test to take longer to execute. I think the correct course of action would be to have this PR increase the timeout in https://github.com/daos-stack/daos/blob/master/src/tests/ftest/telemetry/telemetry_pool_metrics.yaml#L4. I suspect 60 seconds should suffice.

https://github.com/daos-stack/daos/blob/master/src/tests/ftest/telemetry/telemetry_pool_metrics.yaml#L4 looks like the wrong test yaml, that's for a VM test test_telemetry_pool_matrix. instead I should update `daos/src/tests/ftest/telemetry/pool_space_metrics.yaml` to fix the failing test_telemetry_pool_space_metrics right?

**Comment by daltonbohning** [comment]
> https://github.com/daos-stack/daos/blob/master/src/tests/ftest/telemetry/telemetry_pool_metrics.yaml#L4 looks like the wrong test yaml, that's for a VM test test_telemetry_pool_matrix. instead I should update `daos/src/tests/ftest/telemetry/pool_space_metrics.yaml` to fix the failing test_telemetry_pool_space_metrics right?

Yes, that's the correct file to update.

**Review by knard-intel — APPROVED** [review]

**Comment by tanabarr** [comment]
CI run shows TelemetryPoolSpaceMetrics test passing with updated timeout as suggested by @phender , does it look okay to land now or should I run more tests? https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12430/18/testReport/FTEST_telemetry/TelemetryPoolSpaceMetrics/

**Comment by daltonbohning** [comment]
> CI run shows TelemetryPoolSpaceMetrics test passing with updated timeout as suggested by @phender , does it look okay to land now or should I run more tests? https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12430/18/testReport/FTEST_telemetry/TelemetryPoolSpaceMetrics/

Shouldn't that have ran with `Test-nvme: auto_md_on_ssd` as well?

File: src/tests/ftest/server/daos_server_config.yaml
Comment (mjmac) [reply src/tests/ftest/server/daos_server_config.yaml:9]: This comment makes me wonder if we have learned that the default should actually be 16 rather than 8. If we're seeing intermittent failures on hardware tests with the default of 8, then it seems likely to me that the default of 8 is probably going to cause failures out in the field.

We knew from the start of this effort that it was going to involve some experimentation to find defaults that strike the right balance between safety and max utilization of the hardware. As such, I think the testing really should use whatever values we're expecting most customers to use. If those values result in lots of intermittent failures in CI, then that is a canary for what to expect in the field.

**Comment by tanabarr** [comment]
> > CI run shows TelemetryPoolSpaceMetrics test passing with updated timeout as suggested by @phender , does it look okay to land now or should I run more tests? https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12430/18/testReport/FTEST_telemetry/TelemetryPoolSpaceMetrics/
> 
> Shouldn't that have ran with `Test-nvme: auto_md_on_ssd` as well?

Yes I forgot to add that pragma unfortunately, I will hold off pushing again until I am clear as to whether @mjmac wants me to change the default to a higher value.

File: src/tests/ftest/server/daos_server_config.yaml
Comment (tanabarr) [reply src/tests/ftest/server/daos_server_config.yaml:9]: ok, I will increase the default value to 16.

**Review by knard-intel — APPROVED** [review]

**Comment by tanabarr** [comment]
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-12430/19/tests/

Shows the following identified failures:
2 * PoolCreateTests - DAOS-13967,DAOS-13916
DmgSystemReformatTest - DAOS-13613

And 2 new failures:
DmgPoolQueryTest - not specific to this PR, seen on https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/md-on-ssd-testing/63/tests/
SnapshotAggregation - DmgJsonCommandFailure('pool create failed: DER_NOSPACE(-1007): No space on storage target',) looks specific to this PR and possibly to do with the system_ram_reserved default increase to 16gib

**Comment by tanabarr** [comment]
reducing system_ram_reserved to try to make test_snapshot_aggregation pass

**Comment by tanabarr** [comment]
snapshot aggregation test now passing as per https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12430/21/testReport/FTEST_container/SnapshotAggregation/

combining runs 19 & 21 there are only expected failures

identified failures:
2 * PoolCreateTests - DAOS-13967,DAOS-13916
DmgSystemReformatTest - DAOS-13613
DmgPoolQueryTest - DAOS-14072 (PR just created, seen on https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/md-on-ssd-testing/63/tests/)

@phender can you take a look to see if we are at the stage where this could be force landed? obviously pending review approvals

**Review by mjmac — APPROVED** [review]

**Review by kjacque — APPROVED** [review]

**Comment by ashleypittman** [comment]
The ARM CI testing on master failed for this PR, `please fix insufficient ram to meet minimum requirements (mem stats: total 16 GiB (16739012608) - (hugepages 0 B + sys rsvd 16 GiB + (engine rsvd 1.0 GiB * nr engines 1). 4 tgts-per-engine))`

https://github.com/daos-stack/daos/actions/runs/5721566985/job/15503353047

**Pull request merged**
